src/customer/strategies/googlepay/googlepay-authorizenet-customer-strategy.spec.ts

```diff
@@ -0,0 +1,258 @@
+                    expect(button.firstChild).toBe(null);
+                }
+            }
+
+            // Prevent "After Each" failure
+            container = document.createElement('div');
+            document.body.appendChild(container);
+        });
+    });
+
+    describe('#signIn()', () => {
+
+        it('throws error if trying to sign in programmatically', async () => {
+            customerInitializeOptions = getAuthNetCustomerInitializeOptions();
+
+            await strategy.initialize(customerInitializeOptions);
+
+            expect(() => strategy.signIn({ email: 'foo@bar.com', password: 'foobar' })).toThrowError();
+        });
+    });
+
+    describe('#signOut()', () => {
+        beforeEach(async () => {
+            customerInitializeOptions = getAuthNetCustomerInitializeOptions();
+
+            await strategy.initialize(customerInitializeOptions);
+        });
+
+        it('throws error if trying to sign out programmatically', async () => {
+            const paymentId = {
+                providerId: 'googlepayauthorizenet',
+            };
+
+            jest.spyOn(store.getState().payment, 'getPaymentId')
+                .mockReturnValue(paymentId);
+
+            jest.spyOn(remoteCheckoutActionCreator, 'signOut')
+                .mockReturnValue('data');
+
+            const options = {
+                methodId: 'googlepayauthorizenet',
+            };
+
+            await strategy.signOut(options);
+
+            expect(remoteCheckoutActionCreator.signOut).toHaveBeenCalledWith('googlepayauthorizenet', options);
+            expect(store.dispatch).toHaveBeenCalled();
+        });
+
+        it('Returns state if no payment method exist', async () => {
+            const paymentId = undefined;
+            jest.spyOn(store, 'getState');
+
+            jest.spyOn(store.getState().payment, 'getPaymentId')
+                .mockReturnValue(paymentId);
+
+            const options = {
+                methodId: 'googlepayauthorizenet',
+            };
+
+            await strategy.signOut(options);
+
+            expect(store.getState).toHaveBeenCalledTimes(3);
+        });
+    });
+
+    describe('#handleWalletButtonClick', () => {
+        let googlePayOptions: CustomerInitializeOptions;
+
+        beforeEach(() => {
+            googlePayOptions = {
+                methodId: 'googlepayauthorizenet',
+                googlepayauthorizenet: {
+                    container: 'googlePayCheckoutButton',
+                },
+            };
+        });
+
+        it('handles wallet button event', async () => {
+            jest.spyOn(paymentProcessor, 'displayWallet').mockReturnValue(Promise.resolve(getGooglePaymentDataMock()));
+            jest.spyOn(paymentProcessor, 'handleSuccess').mockReturnValue(Promise.resolve());
+            jest.spyOn(paymentProcessor, 'updateShippingAddress').mockReturnValue(Promise.resolve());
+
+            await strategy.initialize(googlePayOptions).then(() => {
```

Comment:
since you are using await, you don't need `.then`

```js
await strategy.initialize(googlePayOptions);
walletButton.click();
expect(paymentProcessor.initialize).toHaveBeenCalled();
```

Reply:
Yep, thanks!